### ISSUE TITLE
Target raises exception when `qubit_properties` is used without passing `num_qubits`

### ISSUE THREAD

**Issue by wshanks** [Qiskit/qiskit#13393]
### Environment

- **Qiskit version**: `main`, like 4c3f8c93037351ff47279c0e2b5211f6bc275e98
- **Python version**: 3.12
- **Operating system**: Fedora Linux

### What is happening?

`Target` raises an exception when `qubit_properties` is used without passing `num_qubits`.

### How can we reproduce the issue?

```python
from qiskit.transpiler import Target, QubitProperties

target = Target(
     qubit_properties=[QubitProperties(frequency=5e9)],
     dt=1e-9,
     granularity=16,
)
print(target.num_qubits)
```

### What should happen?

The example should print `1`. Instead it `Target` raises an exception: `ValueError: The value of num_qubits specified does not match the length of the input qubit_properties list`.

### Any suggestions?

@jakelishman suggested in https://github.com/qiskit-community/qiskit-experiments/pull/1478#discussion_r1828447285 that the `Target` code should be using a default of `None` instead of 0 for `num_qubits` to signal that the number of qubits should be inferred from the qubit properties rather than be set to 0.

**Comment by jakelishman** [comment]
Thanks Will.  Just to add: the same code block worked fine in Qiskit 1.2 and our code clearly _intends_ this to work, so this is a regression in 1.3 and clearly a missing test case from the introduction of the `Target`.

We should get this fixed for 1.3, add the test cases, and ideally rework the documentation of the class arguments a little bit to make it clearer how the `num_qubits` and `qubit_properties` arguments interact.

**Comment by raynelfss** [comment]
This is definitely an oversight in our implementation, I will work on a fix for it.
